planner/core/property_cols_prune.go

```diff
@@ -200,13 +214,22 @@ func (la *LogicalAggregation) PreparePossibleProperties(schema *expression.Schem
 		return nil
 	}
 	resultProperties := make([][]*expression.Column, 0, len(childProps))
+	clonedProperties := make([][]*expression.Column, 0, len(childProps))
 	groupByCols := la.GetGroupByCols()
 	for _, possibleChildProperty := range childProps {
 		sortColOffsets := getMaxSortPrefix(possibleChildProperty, groupByCols)
 		if len(sortColOffsets) == len(groupByCols) {
-			resultProperties = append(resultProperties, possibleChildProperty[:len(groupByCols)])
+			prop := possibleChildProperty[:len(groupByCols)]
+			resultProperties = append(resultProperties, prop)
+			// Clone the Columns in the property before saving them, otherwise the upper Projection may
+			// modify them and lead to unexpected results.
+			clonedProp := make([]*expression.Column, len(prop))
+			for i, col := range prop {
+				clonedProp[i] = col.Clone().(*expression.Column)
+			}
+			clonedProperties = append(clonedProperties, clonedProp)
 		}
 	}
-	la.possibleProperties = resultProperties
-	return la.possibleProperties
+	la.possibleProperties = clonedProperties
```

Comment:
Can we direct use `la.possibleProperties = clonePossibleProperties(resultProperties)` to make the code simpler.

Reply:
That would incur one more loop.